src/common/oclengine.cpp

```diff
@@ -72,6 +84,11 @@ void OCLEngine::InitOCL(int plat, int dev)
     // create the program that we want to execute on the device
     cl::Program::Sources sources;
 
+#if ENABLE_COMPLEX8
+    sources.push_back({ "#define cmplx float2\n#define real1 float\n#define SineShift M_PI_2_F\n", 68UL });
```

Comment:
Can we pull this out into a qcomplex_float.cl and qcomplex_double.cl files, and use the macro at the top of the file to specify what string to load here?

Reply:
Done, with slightly different (shorter) names. I realized something while looking over the CMake for this, though: we need to install the OpenCL kernels with the library, either in a place it can always find them, or pre-compiled as part of the *.a or *.so itself.